include/osquery/killswitch.h

```diff
@@ -0,0 +1,49 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#pragma once
+
+#include <boost/core/noncopyable.hpp>
+#include <osquery/expected.h>
+#include <osquery/status.h>
+#include <string>
+
+namespace osquery {
+
+class Killswitch : private boost::noncopyable {
+ public:
+  enum class SwitchOnError {
+    CallFailed = 1,
+    IncorrectResponseFormat = 2,
+    IncorrectValue = 3
+  };
+
+ private:
+  Killswitch();
+
+ public:
+  ~Killswitch();
+
+  static Killswitch& get() {
+    static Killswitch killswitch;
+    return killswitch;
+  }
+
+  Status refresh();
+
+ private:
+  Expected<bool, SwitchOnError> isSwitchOn(const std::string& key);
```

Comment:
Why single meaningful method of class is `private`?

How can I use it to check something in code? Smt like this:
```c++
if (Killswitch::get().isSwitchOn("name_of_my_super_feature").get_or(false)) {
   // run my feature 
}
```

Reply:
Will add usage examples soon. You should not use isSwitchOn directly. Idea is to add your killswitch function calling isSwitchOn. ( Not to misspell things)